pydantic/utils.py

```diff
@@ -165,3 +166,27 @@ def validate_field_name(bases: List[Type['BaseModel']], field_name: str) -> None
         if getattr(base, field_name, None):
             raise NameError(f'Field name "{field_name}" shadows a BaseModel attribute; '
                             f'use a different field name with "alias=\'{field_name}\'".')
+
+
+@lru_cache(maxsize=None)
+def url_regex_generator(*, relative: bool, require_tld: bool) -> Pattern:
```

Comment:
I'm assuming this is taken from marshmallow?

I think we should add a docstring attributing it to marshmallow with a link to where in code it's taken from.